### PR TITLE
ADBDEV-6041: Migrate ABI tests to Ubuntu 22

### PR DESCRIPTION
**Pull request by silent-observer**
Migrate ABI tests to Ubuntu 22

Now ABI tests in Github Actions fully run on Ubuntu 22. Notable changes:
- Removed dependence on `gcr.io/data-gpdb-public-images/gpdb6-centos7-build`
image (likely maintained by upstream), now building on raw `ubuntu-latest`
from Github.
- Removed custom installation of Universal Ctags, now using package from
Ubuntu repo.
- Building libxerces here because our package is not available for GA.
- Added`-gdwarf-4` GCC flag (needed for abi-checker and GCC 11), which also
enabled source compatibility tests.
- Removed `--with-quicklz` flag since QuickLZ is deprecated and not available
in Ubuntu packages (adding it would require another build step).

### CODE REVIEW

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/79491

File: .github/workflows/greenplum-abi-tests.yml [hunk 1]
```diff
@@ -102,27 +87,35 @@ jobs:
           fetch-depth: 0 # Specify '0' to fetch all history for all branches and tags.
           path: gpdb_src
 
-      - name: Install abi-dumper
+      - name: Install dependencies and abi-dumper
+        run: |
+          sudo gpdb_src/README.ubuntu.bash
+          sudo apt install -y libuv1-dev libpam0g-dev libldap-dev libipc-run-perl abi-dumper
+          sudo ln -fs python2 /usr/bin/python
```
Comment (bandetto) [inline .github/workflows/greenplum-abi-tests.yml:94]: Is sudo required here?

File: .github/workflows/greenplum-abi-tests.yml [hunk 2]
```diff
@@ -102,27 +87,35 @@ jobs:
...
+
+      - name: Build libxerces 3.1.2
         run: |
-          yum install -y epel-release
-          yum install -y abi-dumper
-          yum install -y libzstd-static
+          wget https://archive.apache.org/dist/xerces/c/3/sources/xerces-c-3.1.2.tar.gz
```
Comment (bandetto) [inline .github/workflows/greenplum-abi-tests.yml]: Don't we need our own xerces from Github?

File: .github/workflows/greenplum-abi-tests.yml [hunk 3]
```diff
@@ -102,27 +87,35 @@ jobs:
...
       - name: Build Greenplum
         run: |
           ## TODO: Since abi-dumper requires debug info and it's hard to inject CFLAGS via the script for
           ## releasing Greenplum, we have to manually configure it here. Probably we can improve it in future.
-          export PATH=/opt/python-3.9.13/bin:/opt/python-2.7.18/bin:$PATH
           pushd gpdb_src
           CC='gcc -m64' \
-          CFLAGS='-Og -g3 -Wno-maybe-uninitialized' LDFLAGS='-Wl,--enable-new-dtags -Wl,--export-dynamic' \
-          ./configure --with-quicklz --disable-gpperfmon --with-gssapi --enable-mapreduce --enable-orafce --enable-ic-proxy \
+          CFLAGS='-Og -g3 -gdwarf-4 -Wno-maybe-uninitialized' LDFLAGS='-Wl,--enable-new-dtags -Wl,--export-dynamic' \
```
Comment (bandetto) [inline .github/workflows/greenplum-abi-tests.yml:111]: Isn't dwarf-4 the default debug info type? Is this a requirement for ABI checker?

**Review by bandetto — COMMENTED** [review]
Was this change smoke-tested against intentionally broken ABI already?

File: .github/workflows/greenplum-abi-tests.yml (hunk 1 above)
Comment (silent-observer) [reply .github/workflows/greenplum-abi-tests.yml:94]: Yes, default `ubuntu-latest` container on GA doesn't run as root, so we need sudo to install packages.

File: .github/workflows/greenplum-abi-tests.yml (hunk 2 above)
Comment (silent-observer) [reply .github/workflows/greenplum-abi-tests.yml]: As far as I know, we just need specifically version 3.1.2, so it's acceptable to build it here.

Comment (silent-observer) [reply .github/workflows/greenplum-abi-tests.yml]: I could be wrong, could you explain?

File: .github/workflows/greenplum-abi-tests.yml (hunk 3 above)
Comment (silent-observer) [reply .github/workflows/greenplum-abi-tests.yml:111]: On GCC 11 dwarf-5 is default and is not supported by ABI checker, so switching back to dwarf-4.

**Comment by silent-observer** [comment]
> Was this change smoke-tested against intentionally broken ABI already?

Yes, it actually detected some issues that weren't detected before due to old GCC version (see #1031)

File: .github/workflows/greenplum-abi-tests.yml (hunk 2 above)
Comment (bandetto) [reply .github/workflows/greenplum-abi-tests.yml]: It's needed for ORCA. It's compiled manually here anyway, so you can probably replace it with our one. The repo is there: <https://github.com/arenadata/gp-xerces>.

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/79593

**Review by bandetto — APPROVED** [review]

**Review by whitehawk — APPROVED** [review]

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/79749

**Pull request merged**